scripts/int-log-compiler.sh

```diff
@@ -193,52 +205,65 @@ if [ ! -x "${TEST_BIN}" ]; then
 fi
 
 # start an instance of the simulator for the test, have it use a random port
-SIM_LOG_FILE=${TEST_BIN}_simulator.log
-SIM_PID_FILE=${TEST_BIN}_simulator.pid
-SIM_TMP_DIR=$(mktemp --directory --tmpdir=/tmp tpm_server_XXXXXX)
-PORT_MIN=1024
-PORT_MAX=65534
-BACKOFF_FACTOR=2
-BACKOFF=1
-for i in $(seq 10); do
-    SIM_PORT_DATA=$(od -A n -N 2 -t u2 /dev/urandom | awk -v min=${PORT_MIN} -v max=${PORT_MAX} '{print ($1 % (max - min)) + min}')
-    SIM_PORT_CMD=$((${SIM_PORT_DATA}+1))
-    echo "Starting simulator on port ${SIM_PORT_DATA}"
-    simulator_start ${SIM_BIN} ${SIM_PORT_DATA} ${SIM_LOG_FILE} ${SIM_PID_FILE} ${SIM_TMP_DIR}
-    sleep 1 # give daemon time to bind to ports
-    PID=$(cat ${SIM_PID_FILE})
-    echo "simulator PID: ${PID}";
-    netstat -ltpn 2> /dev/null | grep "${PID}" | grep -q "${SIM_PORT_DATA}"
-    ret_data=$?
-    netstat -ltpn 2> /dev/null | grep "${PID}" | grep -q "${SIM_PORT_CMD}"
-    ret_cmd=$?
-    if [ \( $ret_data -eq 0 \) -a \( $ret_cmd -eq 0 \) ]; then
-        echo "Simulator with PID ${PID} bound to port ${SIM_PORT_DATA} and " \
-             "${SIM_PORT_CMD} successfully.";
-        break
-    fi
-    echo "Port conflict? Cleaning up PID: ${PID}"
-    kill "${PID}"
-    BACKOFF=$((${BACKOFF}*${BACKOFF_FACTOR}))
-    echo "Failed to start simulator: port ${SIM_PORT_DATA} or " \
-         "${SIM_PORT_CMD} probably in use. Retrying in ${BACKOFF}."
-    sleep ${BACKOFF}
-    if [ $i -eq 10 ]; then
-        echo "Failed to start simulator after $i tries. Giving up.";
-        exit 1
-    fi
-done
+if [ -n "${SIM_BIN}" ]; then
```

Comment:
I think you may be better off using a separate script for this. The current `int-log-compiler.sh` handles all the magic around running the simulator binary for each test case. If you're executing against the hardware TPM there's no need for any of that magic any longer. In `Makefile.am` you can change the log compiler for a given file extension by setting the `INT_LOG_COMPILER` variable to a different script. This will be a lot cleaner since the different implementations will be in separate scripts.

I think this script can be pretty simple if you only run a single instance of the daemon with all tests executed against it. This should allow you to run all of the tests in parallel (something the daemon should be able to handle). You will however have to deal with whatever elevated permissions are required to run the daemon, as well as the possibility that an instance of the daemon may already be running.